lib/sidekiq_base.cc

```diff
@@ -274,86 +280,127 @@ float sidekiq_base<HdlType>::read_temperature() {
-	bool result{true};
+int sidekiq_base<HdlType>::start_streaming() {
+    int status;
+    if (dual_channel) {
+        status = sidekiq_functions.start_streaming_func(card, hdl2);
+        if (status != 0) {
+            printf("Error: could not start streaming, status %d, %s\n", status, strerror(abs(status)) );
+            exit(status);
+        }
+    }
 
-	if (sidekiq_functions.start_streaming_func(card, hdl) != 0) {
-		printf("Error: could not start streaming\n");
-		result = false;
+	status = sidekiq_functions.start_streaming_func(card, hdl);
+	if (status != 0) {
+		printf("Error: could not start streaming, status %d, %s\n", status, strerror(abs(status)) );
+        exit(status);
 	}
-	return result;
+	return status;
 }
 
 template<typename HdlType>
-bool sidekiq_base<HdlType>::stop_streaming() {
-	bool result{true};
+int sidekiq_base<HdlType>::stop_streaming() {
+    int status;
+    if (dual_channel) {
+        status = sidekiq_functions.stop_streaming_func(card, hdl2);
+        if (status != 0) {
+            printf("Error: could not stop streaming, status %d, %s\n", status, strerror(abs(status)) );
+        }
+    }
 
-	if (sidekiq_functions.stop_streaming_func(card, hdl) != 0) {
-		printf("Error: could not stop streaming\n");
-		result = false;
+	status = sidekiq_functions.stop_streaming_func(card, hdl);
+	if (status != 0) {
+		printf("Error: could not stop streaming, status %d, %s\n", status, strerror(abs(status)) );
 	}
-	return result;
+	return status;
 }
 
 template<typename HdlType>
-double sidekiq_base<HdlType>::get_sample_rate() {
+double sidekiq_base<HdlType>::get_sample_rate(HdlType handle) {
+    int status;
 	uint32_t p_rate;
 	double p_actual_rate;
 	uint32_t p_bandwidth;
 	uint32_t p_actual_bandwidth;
 
-	if (sidekiq_functions.get_sample_rate_func(card, hdl, &p_rate, &p_actual_rate, &p_bandwidth, &p_actual_bandwidth) !=
-		0) {
-		printf("Error: failed to set sample rate or bandwidth\n");
+	status = sidekiq_functions.get_sample_rate_func(card, handle, &p_rate, &p_actual_rate, &p_bandwidth, &p_actual_bandwidth);
+	if (status != 0) {
+		printf("Error: failed to get sample rate, status %d, %s\n", status, strerror(abs(status)) );
+        exit(status);
 	}
 	return p_actual_rate;
 }
 
 template<typename HdlType>
-bool sidekiq_base<HdlType>::set_samplerate_bandwidth(double sample_rate, double bandwidth) {
-	bool result{true};
+int sidekiq_base<HdlType>::set_samplerate_bandwidth(double sample_rate, double bandwidth) {
+    int status;
 	auto rate = static_cast<uint32_t>(sample_rate);
 	auto bw = static_cast<uint32_t>(bandwidth);
 
-	if (sidekiq_functions.set_sample_rate_func(card, hdl, rate, bw) != 0) {
-		printf("Error: could not set sample_rate %f and bandwidth %f\n", sample_rate, bandwidth);
-		result = false;
+
+	status = sidekiq_functions.set_sample_rate_func(card, hdl, rate, bw);
+	if (status != 0) {
+		printf("Error: could not set sample_rate, status %d, %s\n", status, strerror(abs(status)) );
+        exit(status);
```

Comment:
same question about calling exit here...not sure that's what you want to do

Reply:
Ok this is a set function, so I removed the exits and just return the status.